src/parser/xmlHandler.js

```diff
@@ -162,8 +164,8 @@ class XMLHandler {
         }
       }
 
-      if (val == null) {
-        if (descriptor.isNillable) {
+      if (val === null ||(val === undefined && this.options.ignoreAttributesUndefined == true)) {
+        if (descriptor.isNillable ) {
```

Comment:
Though about this for a while; The naming is a little unintuitive.

My suggestion is `jsonUndefinedAsXmlNil`:

1. It informs the flow of the conversion (JSON->XML)
2. It clearly states what it does (Coerce JSON `undefined` as XML `xsi:nil`)

The only thing missing is stating that the XML element must be defined as `nillable`, but that can be covered by documentation.

```suggestion
      if (val === null ||(val === undefined && this.options.jsonUndefinedAsXmlNil == true)) {
        if (descriptor.isNillable) {
```